integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/massindexing/MassIndexingCachingIT.java

```diff
@@ -150,19 +166,76 @@ public void explicit_get() {
 		assertSoftly( softly -> {
 			assertEntityLoadCount( softly ).isEqualTo( 1 );
 			assertSecondLevelCacheHitCount( softly ).isEqualTo( 2 );
+			assertSecondLevelCachePutCount( softly ).isEqualTo( 0 );
+		} );
+	}
+
+	@Test
+	public void explicit_normal() {
+		setupHolder.runNoTransaction( session -> {
+			SearchSession searchSession = Search.session( session );
+			MassIndexer indexer = searchSession.massIndexer()
+					.cacheMode( CacheMode.NORMAL );
+
+			backendMock.expectWorks( IndexedEntity.NAME, DocumentCommitStrategy.NONE, DocumentRefreshStrategy.NONE )
+					.add( "1", b -> b.field( "text", "text1" ) )
+					.add( "2", b -> b.field( "text", "text2" ) )
+					.add( "3", b -> b.field( "text", "text3" ) );
+
+			// purgeAtStart and mergeSegmentsAfterPurge are enabled by default,
+			// so we expect 1 purge, 1 mergeSegments and 1 flush calls in this order:
+			backendMock.expectIndexScaleWorks( IndexedEntity.NAME, session.getTenantIdentifier() )
+					.purge()
+					.mergeSegments()
+					.flush()
+					.refresh();
+
+			try {
+				indexer.startAndWait();
+			}
+			catch (InterruptedException e) {
+				fail( "Unexpected InterruptedException: " + e.getMessage() );
+			}
+
+		} );
+
+		backendMock.verifyExpectationsMet();
+
+		assertSoftly( softly -> {
+			assertEntityLoadCount( softly ).isEqualTo( 1 );
+			assertSecondLevelCacheHitCount( softly ).isEqualTo( 2 );
+			assertSecondLevelCachePutCount( softly ).isEqualTo( 1 );
 		} );
 	}
 
-	public AbstractLongAssert<?> assertEntityLoadCount(SoftAssertions softly) {
+	private AbstractLongAssert<?> assertEntityLoadCount(SoftAssertions softly) {
 		return softly.assertThat( statistics.getEntityLoadCount() )
 				.as( "Entity load count" );
 	}
 
-	public AbstractLongAssert<?> assertSecondLevelCacheHitCount(SoftAssertions softly) {
+	private AbstractLongAssert<?> assertSecondLevelCacheHitCount(SoftAssertions softly) {
 		return softly.assertThat( statistics.getSecondLevelCacheHitCount() )
 				.as( "Second level cache hit count" );
 	}
 
+	private AbstractLongAssert<?> assertSecondLevelCachePutCount(SoftAssertions softly) {
+		return softly.assertThat( statistics.getSecondLevelCachePutCount() )
+				.as( "Second level cache put count" );
+	}
+
+	private Query cachedQuery(Session session, CacheMode cacheMode) {
+		Query<IndexedEntity> query = session.createQuery(
+				"select e from IndexedEntity e where e.id in (:ids)",
+				IndexedEntity.class
+		);
+
+		query.setParameter( "ids", Arrays.asList( 1, 2, 3 ) )
+				.setCacheMode( cacheMode )
+				.setHint( QueryHints.HINT_CACHEABLE, true );
```

Comment:
This shouldn't be necessary? We are not supposed to rely the query cache anymore.

Reply:
You're right. I seems that we don't need it anymore!